src/Symfony/Bundle/FrameworkBundle/Controller/TemplateController.php

```diff
@@ -12,17 +12,48 @@
 namespace Symfony\Bundle\FrameworkBundle\Controller;
 
 use Symfony\Component\DependencyInjection\ContainerAwareInterface;
-use Symfony\Component\DependencyInjection\ContainerAwareTrait;
+use Symfony\Component\DependencyInjection\ContainerInterface;
 use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Templating\EngineInterface;
+use Twig\Environment;
 
 /**
  * TemplateController.
  *
  * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @final since version 3.4
  */
 class TemplateController implements ContainerAwareInterface
 {
-    use ContainerAwareTrait;
+    /**
+     * @deprecated since version 3.4, to be removed in 4.0
+     */
+    protected $container;
+
+    private $templating;
+    private $twig;
+
+    public function __construct(EngineInterface $templating = null, Environment $twig = null)
```

Comment:
I would put Twig first as the other one will be deprecated.

Reply:
done